meteor/client/ui/Status/UserActivity.tsx

```diff
@@ -26,11 +26,62 @@ interface IUserActionsListProps {
 	renderButtons?: (item: UserActionsLogItem) => React.ReactElement
 }
 
-function prettyPrintJsonString(str: string): string {
-	try {
-		return JSON.stringify(JSON.parse(str), undefined, 4)
-	} catch (_e) {
-		return str
+/**
+ * pretty-prints JSON content, and collapses it if it's too long.
+ */
+function CollapseContent(props: { content: any }): JSX.Element {
+	const [expanded, setExpanded] = useState(false)
+
+	const originalString = useMemo(() => {
+		try {
+			let str = JSON.stringify(JSON.parse(props.content), undefined, 2) // Pretty print JSON
```

Comment:
I was going to say something here about potentially trying to avoid an unnecessary `[` and `]` wrapping on this logging, but that is probably better handled where it is being generated. I'll look into that

Reply:
https://github.com/nrkno/sofie-core/pull/1245